doc/tutorials/basic/serve.md

```diff
@@ -0,0 +1,181 @@
+- `serve`: the command you want Panel to run
+- `app.py`: the file `app.py` you want to serve
+- `--autoreload`: make the server restart after code changes. Use this for **development only**.
+:::
+
+In the output, you will find the line
+
+```bash
+Bokeh app running at: http://localhost:5006/app
+```
+
+That line shows the URL where your app is being served, on your local machine.
+
+:::{note}
+The `Bokeh server` is mentioned because Panel is built on top of [Bokeh](https://docs.bokeh.org).
+:::
+
+Open your browser at [http://localhost:5006/app](http://localhost:5006/app).
+
+The application will look like.
+
+![Panel serve single .py file](../../_static/images/panel-serve-py-app.png).
+
+Now stop the server by pressing `CTRL+C` one or more times in the terminal.
+
+## Serve a notebook
+
+Copy the 2 code cells below into a clean notebook named `app.ipynb`.
+
+```python
+import panel as pn
+
+pn.extension()
+```
+
+```python
+pn.panel("Hello Notebook World").servable()
+```
+
+Run the cells in the notebook and save it as `app.ipynb` if you have not already done it.
+
+It should look like
+
+![Panel Notebook App](../../_static/images/panel-serve-ipynb-notebook.png).
+
+:::{note}
+The code refers to
+
+- `panel`: The Panel python package. It's a convention to import it as `pn`.
+- `pn.extension()`: **Loads the [`pyviz_comms`](https://github.com/holoviz/pyviz_comms) notebook extension**, loads javascript dependencies and configures Panel.
+- `pn.panel(...)`: Creates a *displayable* Panel component. **The component can be displayed directly in the notebook**.
+- `.servable()`: Displays the component in a *server app*.
+:::
+
+Run the Panel server in your terminal with
+
+```bash
+panel serve app.ipynb --autoreload
+```
+
+When the server has started, the terminal output should look like this
+
+```bash
+$ panel serve app.ipynb --autoreload
+2024-01-17 21:05:32,338 Starting Bokeh server version 3.3.3 (running on Tornado 6.4)
+2024-01-17 21:05:32,339 User authentication hooks NOT provided (default user enabled)
+2024-01-17 21:05:32,342 Bokeh app running at: http://localhost:5006/app
+2024-01-17 21:05:32,342 Starting Bokeh server with process id: 42008
+```
+
+:::{note}
+The command `panel serve app.ipynb --autoreload` refers to:
+
+- `panel`: the panel executable.
+- `serve`: the command you want Panel to run
+- `app.ipynb`: the file `app.ipynb` you want to serve
+- `--autoreload`: make the server restart after code changes. Use this for **development only**.
+:::
+
+In the output, you will find the line
+
+```bash
+Bokeh app running at: http://localhost:5006/app
+```
+
+That line shows the URL where the app is being served, on the local machine.
+
+Open your browser at [http://localhost:5006/app](http://localhost:5006/app).
+
+The application will look like.
+
+![Panel serve single .ipynb file](../../_static/images/panel-serve-ipynb-app.png).
+
+Now stop the server by pressing `CTRL+C` one or more times in the terminal.
```

Comment:
Feels redundant; can you simply link above so there's only one source of truth?

Reply:
I've tried another approach I would like to test first. Using tabs. There is now a tab for *Python Script*, *Notebook* and *Markdown*.

Python script users and notebooks users can be really, really different. I feel like providing an example for one persona and not the other we will miss out on 50% of users.

Reply:
Tabs make sense to me.

Reply:
I tabs work out here. I will also try applying it to the `pn.panel` and `panes` section so that there is a tab for each of the most popular plotting libraries instead of a section for each now. Then users can focus on the plotting libraries they are interested in.

Reply:
Tabs are now there https://holoviz-dev.github.io/panel/tutorials/basic/serve.html. I think it works.

And I think its an excellent solution for us. We have the problem that we are very flexbility.

- Support pip and conda
- Support .py, .ipynb and .md files
- Support Altair, ECharts, hvPlot, HoloViews, Matplotlib, Plotly, ...

Sometimes picking one of this means we leave 80% of users behind. So for key use cases tabs can specific guidance to users without seeming overwhelming.

https://github.com/holoviz/panel/assets/42288570/2202b84b-ee5b-433b-a287-818abccca849

I will close this one. Feel free to open again if you think this does not address your issue.